cmd/login/runner.go

```diff
@@ -205,13 +215,11 @@ func (r *runner) loginWithCodeName(ctx context.Context, codeName string) error {
 		}
 
 		return nil
-	} else if err != nil {
-		return microerror.Mask(err)
 	}
 
 	if contextAlreadySelected {
 		fmt.Fprintf(r.stdout, "Context '%s' is already selected.\n", contextName)
-	} else {
+	} else if !(r.flag.KeepContext || selfContained) {
 		fmt.Fprintf(r.stdout, "Switched to context '%s'.\n", contextName)
```

Comment:
TODO: go through all of these info statements and add the needed information that a self-contained kubeconfig was created and also make sure there are no inaccurate "switch context" statements anymore.